### PR TITLE
Refactor bootstrap buttons into one file using styled components

### PR DESCRIPTION
**Pull request by pmyip3**
# Description 
Refactor bootstrap buttons into one file using styled-components
Change the sidebar color 

# Type of change 
- [ ] Bug fix
- [ ] New feature
- [x] Refactoring
- [ ] Documentation 
- [x] Improvement 

# Problem
The button style and color isn't consistent in Researcify, also it would be hard to change the button individually to make them consistent

# Solution description
Shared styled buttons are created inside the client/shared/components, we just need to modify the styled button if we want to change the button style and color for the entire website. 

Currently, there are purple, gold buttons also delete buttons implemented, feel free to create your own styled button or change the color to see if it has a better UI.  

# Tests 
All unit tests pass? N/A

Builds successfully? Yes

Attach screenshot of the expected result (e.g. UI of the website/ postman api call result/ database changes)
image

Provide steps in detail for reviewers to reproduce the expected result of the code changes.

# Relevant document/ link (if any) 
https://styled-components.com/

# Risk
Low. UI modification only

### CODE REVIEW

**Review by ssha0054 — APPROVED** [review]
Really nice PR! The `styledComponents` file is a great addition and makes it so much easier to use existing styles and create new ones. Decided to follow this approach with the changes we were making on the Achievements page after seeing this PR go up and it made it pretty easy to restyle the buttons.

File: client/src/components/dashboard/deploy/DeployPage.js [hunk 1]
```diff
@@ -37,7 +66,7 @@ const DeployPage = ({ teamId }) => {
   };
 
   const GitHubLoginButton = (
-    <GitHubLogin
+    <GHButton
       className="float-right github-login-button"
```
Comment (jfan0010) [inline client/src/components/dashboard/deploy/DeployPage.js:70]: do we still need the `github-login-buton` here? cause if I remember correctly, that's the styling of the previous button (ie. change the colour to blue etc.)
If we don't need this maybe we can also remove the CSS part of `github-login-button`

File: client/src/components/shared/styledComponents.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    padding: .375rem .75rem;
+    border: 1px solid #56658a;
+    border-radius: .25rem;
+    background-color: transparent;
+    color: #56658a;
+    transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
+    font-size: ${(props) => props.fontSize}; //.875rem;
+  &:hover{
+    background-color:rgb(60, 70, 96);
+    color: white;
```
Comment (jfan0010) [inline client/src/components/shared/styledComponents.js:68]: Personally has a feeling that these stylings are a bit overwhelming 🙈.  Not sure if there is a way to refactor them, but I think for now they are alright.

**Review by jfan0010 — APPROVED** [review]
LGTM, the new theme for Researchify actually looks nice to me!

**Review by vgoh1 — APPROVED** [review]

**Pull request merged**